components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/inbound/InboundMessageProcessor.kt

```diff
@@ -80,7 +80,14 @@ internal class InboundMessageProcessor(
                     if (err == null) {
                         publisher.publish(listOf(ack))
                     } else {
-                        logger.info("Failed to publish ack message.", err)
+                        val lastIem = traceable.item.records.lastOrNull()
+                        val topic = lastIem?.topic
+                        val message = (lastIem?.value as? AppMessage)?.message as? AuthenticatedMessage
+                        logger.info(
+                            "Failed to publish message ${message?.header?.messageId} to the '$topic' topic. " +
```

Comment:
```suggestion
                            "Failed to publish message '${message?.header?.messageId}' to the '$topic' topic. " +
```
just a nit, otherwise LGTM